File: C++/find-the-highest-altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int altitude = 0;
        int maxAltitude = 0;
        for (int g: gain) {
            altitude += g;
            if (maxAltitude < altitude) {
                maxAltitude = altitude;
            }
        }
        return maxAltitude;
    }
};

File: TypeScript/find-the-highest-altitude.ts
function largestAltitude(gain: number[]): number {
    let altitude = 0
    let highest = 0
    for (let i = 0; i < gain.length; i++) {
        altitude += gain[i]
        if (highest < altitude) {
            highest = altitude
        }
    }
    return highest
};